### PR TITLE
Schema 4.6 draft

### PR DESCRIPTION
**Pull request by ashwinisukale**
## Purpose

closes: https://github.com/datacite/schema/issues/141

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)

- [ ] New feature (non-breaking change which adds functionality)

- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Reviewer, please remember our [guidelines](https://datacite.atlassian.net/wiki/spaces/TEC/pages/1168375809/Pull+Request+Guidelines):

- Be humble in the language and feedback you give, ask don't tell.
- Consider using positive language as opposed to neutral when offering feedback. This is to avoid the negative bias that can occur with neutral language appearing negative.
- Offer suggestions on how to improve code e.g. simplification or expanding clarity.
- Ensure you give reasons for the changes you are proposing.

### CODE REVIEW

**Review by codycooperross — CHANGES_REQUESTED** [review]
This is a great start, @ashwinisukale , but can you also add tests for the new `relatedIdentifierType`, `contributorType`, `relationType`, and `dateType` values? See the summary of changes here: https://docs.google.com/document/d/1RllY4Q-_Ls9gHSfy9Iuwlctv0uV4E_S6Sg56rvkMt1o/edit#heading=h.76dlmbn6ivq4

File: spec/kernel-4.5/other_spec.rb
Comment (svogt0511) [reply spec/kernel-4.5/other_spec.rb:38]: @codycooperross - Regarding your comment on 'let(:root)...' it looks like the other files in other kernel-4.x directories use kernel-4.  Should this be changed?

**Review by svogt0511 — COMMENTED** [review]
How does this release play with the stylistic changes that Kelly is doing?  Are they being released at the same time?

**Comment by codycooperross** [comment]
> How does this release play with the stylistic changes that Kelly is doing? Are they being released at the same time?

@KellyStathis I'm not sure I have the most context on this—can you provide some insight?

File: spec/kernel-4.5/other_spec.rb
Comment (codycooperross) [reply spec/kernel-4.5/other_spec.rb:38]: Probably, although I'm not really sure! These may have remained unchanged because the tests were unaffected by later schema changes. If it's not too much of a burden to change, it may be beneficial. That said, the test in `spec/kernel-4/other_spec.rb` should obviously point to the `source/meta/kernel-4` directory.

File: spec/kernel-4.6/example_with_full_spec.rb
Comment (ashwinisukale) [inline spec/kernel-4.6/example_with_full_spec.rb:1]: @codycooperross @svogt0511 

I have added most of the test cases in this file, instead of having multiple files for testing different part of the XML. I will create a issue which will clean all the tests inside schema. Right now they are all having incorrect file path.

File: spec/kernel-4.6/other_spec.rb
Comment (svogt0511) [inline spec/kernel-4.6/other_spec.rb:1]: 2nd ResourceTypeGeneral in other_spec.rb should have 'Project' instead of 'Award'?

File: spec/kernel-4.5/other_spec.rb
Comment (svogt0511) [reply spec/kernel-4.5/other_spec.rb:38]: Definitely award and project should not be in this test.  It passes because it uses the latest 4.x schema and not 4.5.

File: spec/kernel-4.6/example_with_full_spec.rb
Comment (svogt0511) [reply spec/kernel-4.6/example_with_full_spec.rb:1]: Looks good.  The only one I don't see is 'dateType: Coverage'.  That may be elsewhere?

**Comment by KellyStathis** [comment]
> > How does this release play with the stylistic changes that Kelly is doing? Are they being released at the same time?
> 
> @KellyStathis I'm not sure I have the most context on this—can you provide some insight?

Sorry @codycooperross and @svogt0511 I missed this last week.

The redesign of the schema site is on track to launch at the end of October. Once that's complete, I would merge those changes into this `schema-4.6-draft` branch. It is mostly CSS changes, but some of the pages will have minor structural changes (e.g., changing a heading level) that we'll want to incorporate here.

File: source/meta/kernel-4/index.html.md
Comment (KellyStathis) [reply source/meta/kernel-4/index.html.md:1]: I think these FIXMEs are mostly release dates since this is still TBD. I have a task in the [release plan](https://docs.google.com/document/d/1FicisbNhc3XJrm2cBUdehaLdYgxmgXuFQKvI3T9-j2o/edit?tab=t.0#heading=h.c6i7e14shtld) to correct these.

File: spec/kernel-4.6/example_with_full_spec.rb
Comment (ashwinisukale) [reply spec/kernel-4.6/example_with_full_spec.rb:1]: Thats was a good catch, adding it

Comment (ashwinisukale) [reply spec/kernel-4.6/example_with_full_spec.rb:1]: I just added all the date types and tested its presence.

**Review by codycooperross — COMMENTED** [review]
I think one of the last pieces of this work, assuming that all the new values are tested as valid against the XSD, is to copy all of the `spec/kernel-4.6` tests to the `spec/kernel-4` folder. These two should be identical, to my memory, because the XSD is identical as well. @svogt0511 does that sound right?

**Comment by ashwinisukale** [comment]
> I think one of the last pieces of this work, assuming that all the new values are tested as valid against the XSD, is to copy all of the `spec/kernel-4.6` tests to the `spec/kernel-4` folder. These two should be identical, to my memory, because the XSD is identical as well. @svogt0511 does that sound right?

I have created a ticket which will address this https://github.com/datacite/schema/issues/146 here I am also addressing correction of test cases in other versions.

**Review by svogt0511 — APPROVED** [review]
As far as I can see my comments have been addressed.  I am going to approve this pending resolution of Cody's comments.  I assume that this is not going to be released until @KellyStathis is ready to make the announcement of 4.6, and at that time, everything related to 4.6 will be released.  I am also assuming that 4.6 is scheduled to go after the release of the site format/css changes.

**Review by KellyStathis — APPROVED** [review]

**Review by codycooperross — APPROVED** [review]

**Review by codycooperross — APPROVED** [review]

**Pull request merged**
